airbyte-cdk/python/airbyte_cdk/sources/streams/http/http.py

```diff
@@ -171,13 +196,27 @@ def backoff_time(self, response: requests.Response) -> Optional[float]:
         return None
 
     def _create_prepared_request(
-        self, path: str, headers: Mapping = None, params: Mapping = None, json: Any = None
+        self,
+        path: str,
+        headers: Mapping = None,
+        params: Mapping = None,
+        json: Any = None,
+        data: Any = None
     ) -> requests.PreparedRequest:
-        args = {"method": self.http_method, "url": self.url_base + path, "headers": headers, "params": params}
-
-        if self.http_method.upper() == "POST":
-            # TODO support non-json bodies
-            args["json"] = json
+        args = {
+            "method": self.http_method,
+            "url": self.url_base + path,
+            "headers": headers,
+            "params": params
+        }
+        if self.http_method.upper() in BODY_REQUEST_METHODS:
+            if json and data:
+                raise RequestBodyException("At the same time only one of the 'request_body_data' and "
```

Comment:
The message of the exception could be put in the __init__ of the exception class, and here we only raise the exception, assuming this message doesn't change from request to request.

Reply:
yes, but developers will be able to implement more complicated logic for body functions depending on input attributes. Thus I propose to check that only one body function returns some payload at the same time.